packages/sanity/src/core/create/context/SanityCreateConfigProvider.tsx

```diff
@@ -0,0 +1,35 @@
+import {type ReactNode, useMemo} from 'react'
+import {SanityCreateConfigContext} from 'sanity/_singletons'
+
+import {useSource} from '../../studio'
+import {CreateLinkedActions} from '../components/CreateLinkedActions'
+import {CreateLinkedDocumentBanner} from '../components/CreateLinkedDocumentBanner'
+import {type SanityCreateConfigContextValue} from './useSanityCreateConfig'
+
+interface SanityCreateConfigProviderProps {
+  children: ReactNode
+}
+
+/**
+ * @internal
+ */
+export function SanityCreateConfigProvider(props: SanityCreateConfigProviderProps): JSX.Element {
+  const {children} = props
+  const {beta} = useSource()
+  const value = useMemo((): SanityCreateConfigContextValue => {
+    return {
+      ...beta?.create,
+      startInCreateEnabled: !!beta?.create?.startInCreateEnabled,
+      components: {
```

Comment:
I think my only concern about this is what I have commented previously about the banners.

Technically you could also export these directly from the sanity package. Is there a reason why that wasn't considered vs this approach? 🤔

Reply:
I got to admit , I did not consider that approach. I've tried to limit what gets exported, as not to pollute the global `sanity´ api, was the thought. Therefore, only the props are exported, ie the interface. The implementations are internal.

I'd be happy to export the components, and ditch the context approach if you prefer. 
To me, it just seems non-idiomatic to keep implementations without an interface in one package and use directly in another. I realize now that the way I'm going about this might break established `sanity` conventions though, so I'm happy to take your lead on this.

Reply:
I don't think there is a conventional way per se of doing things; think of it more as a "mindless way that we've been going about it" than anything intentional :)

I just wanted to double check. I don't have a strong opinion about the way of exporting in particular and I'd be ok with this approach :)